projects/fw_103/src/fw_103.c

```diff
@@ -0,0 +1,61 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// interrupt callback function
+static void prv_button_pressed(const GpioAddress *address, void *context) {
+  GpioAddress *adc_read = (GpioAddress *)context;
+  uint16_t adc_value = 0;
+  adc_read_converted_pin(*adc_read, &adc_value);
+  LOG_DEBUG("%d\n", adc_value);
+}
+
+int main(void) {
+  interrupt_init();
+  soft_timer_init();
+  gpio_init();
+  gpio_it_init();
+
+  GpioAddress adc_reading_addr = { .port = GPIO_PORT_A, .pin = 6 };
+  GpioAddress button_addr = { .port = GPIO_PORT_B, .pin = 2 };
+
+  GpioSettings adc_settings = {
+    .direction = GPIO_DIR_IN,
+    .state = GPIO_STATE_LOW,
+    .resistor = GPIO_RES_NONE,
+    .alt_function = GPIO_ALTFN_ANALOG,
+  };
+
+  GpioSettings button_settings = {
+    .direction = GPIO_DIR_IN,
+    .state = GPIO_STATE_LOW,
+    .resistor = GPIO_RES_NONE,
+    .alt_function = GPIO_ALTFN_ANALOG,
```

Comment:
You won't need to initialize the button with `GPIO_ALTFN_ANALOG` since you aren't reading from the button pin with the ADC, it can just be `GPIO_ALTFN_NONE`.